test/mpi/main.cc

```diff
@@ -3,19 +3,29 @@
 #include "test_mpi.h"
 #include <chrono>
 
+int test_rand_seed = -1;
+size_t test_max_size = 0;
```

Comment:
maybe make "max_size" part of the test class.
Also, lets make non 0 default, say 8 Mb. Then, e.g. running on the test on 1K ranks, will mean we can still run allgather up to 8K msg size per proc.

Reply:
fixed